src/core/tsi/alts/handshaker/alts_handshaker_client.cc

```diff
@@ -732,16 +732,14 @@ alts_handshaker_client* alts_grpc_handshaker_client_create(
   client->handshake_status_details = grpc_empty_slice();
   client->max_frame_size = max_frame_size;
   client->error = error;
-  grpc_slice slice = grpc_slice_from_copied_string(handshaker_service_url);
```

Comment:
It may be slightly tricky to remove the `handshaker_service_url` parameter since it's used on the next line:
```
 client->call = strcmp(handshaker_service_url, ALTS_HANDSHAKER_SERVICE_URL_FOR_TESTING) == 0 ? nullptr : ...
```
If this the `strcmp` is removed and `grpc_channel_create_pollset_set_call(...)` is always used instead, then we need to refactor the unit tests, which rely on this line, I think. If this is something you'd like to be done long-term, please file a bug and assign to me and I can clean it up when I have cycles.

Reply:
@matthewstevenson88 IIUC `handshaker_service_url ` is not removed from the parameter list of `alts_grpc_handshaker_client_create` so it still can be used in `strcmp` LOC. The commit removes it only from `grpc_channel_create_pollset_set_call ` (replaces with `nullptr`) so `alts_handshaker_client.cc` looks fine.

However I couldn't find relevant tests across the codebase for `grpc_channel_create_pollset_set_call(...)` to make sure that authority will be calculated correctly. Could anybody point me in the right direction?

Reply:
One idea to test this:

We could start up the [fake ALTS handshaker service](https://github.com/grpc/grpc/blob/master/test/core/tsi/alts/fake_handshaker/fake_handshaker_server.cc) (an in-process gRPC server), and perform an ALTS handshake, setting the handshaker service url to something that needs this PR. We could add a field to the ALTS handshake server to check the RPC authority with [ExperimentalGetAuthority()](https://github.com/grpc/grpc/blob/c8407f8ef955ec064119790f41eb6e0469cc2df1/include/grpcpp/server_context.h#L300)

Reply:
As pointed out already, removing `handshaker_service_url` cannot be done without significantly refactoring the unit tests. IMHO such a refactor should not block this, so resolving. FWIW, I am not very familiar with the code but did also look for a good way to test that the correct authority is selected, but found nothing all that easy.